packages/KDateTimePicker/KDateTimePicker.vue

```diff
@@ -0,0 +1,562 @@
+        end: { class: 'vcal-day-end' }
+      }
+    }
+    const calendarDragAttributes = {
+      highlight: {
+        start: { class: 'vcal-day-drag-start' },
+        base: { class: 'vcal-day-drag-base' },
+        end: { class: 'vcal-day-drag-end' }
+      }
+    }
+    const hasCalendar = computed(() => props.mode !== 'relative')
+    const hasTimePeriods = computed(() => props.timePeriods && props.timePeriods.length)
+    const showCalendar = computed(() => tabName.value === 'custom' || !hasTimePeriods.value)
+    const hasRelativeTimeframes = computed(() => props.timePeriods.length > 0)
+    const submitDisabled = computed(() => {
+      // If either the calendar is in range selection mode, or relative time frames
+      // are present, check whether both `start` and `end` are set;
+      // Otherwise, it's a single date or time, so only check `start`
+      return props.range || hasRelativeTimeframes.value
+        ? !selectedRange.value.start || !selectedRange.value.end
+        : !selectedRange.value.start
+    })
+    const defaultTimeframe = computed(() => {
+      return props.defaultRelative !== undefined
+        ? props.defaultRelative
+        : null
+    })
+
+    const changeCalendarRange = (vCalValue) => {
+      let start = ''
+      let end = ''
+
+      // If value is an object, this is a time range. Else, a single date or time value.
+      if (vCalValue.hasOwnProperty('start')) {
+        start = vCalValue.start
+        end = vCalValue.end
+      } else {
+        start = vCalValue
+      }
+
+      // Set emitted value when v-calendar selection is made
+      selectedRange.value = {
+        start: new Date(start),
+        end: new Date(end),
+        timePeriodsKey: ''
+      }
+    }
+
+    /**
+     * Updates both the input field value, and the full time frame readout
+     * when a relative time frame button is clicked
+     * @param {*} timeframe
+     */
+    const changeRelativeTimeframe = (timeframe) => {
+      selectedTimeframe.value = timeframe
+
+      // Format the start/end values as human readable date
+      const start = selectedTimeframe.value.start()
+      const end = selectedTimeframe.value.end()
+
+      // Set value to be emitted when relative time frame clicked
+      selectedRange.value = {
+        start: new Date(start),
+        end: new Date(end),
+        timePeriodsKey: selectedTimeframe.value.key
+      }
+
+      fullRangeDisplay.value = formatDisplayDate({ start, end })
+    }
+
+    const clearSelection = () => {
+      selectedCalendarRange.value = null
+      abbreviatedDisplay.value = props.placeholder
+      fullRangeDisplay.value = ''
+      selectedRange.value = { start: '', end: '', timePeriodsKey: '' }
+      selectedTimeframe.value = props.timePeriods[0]
+
+      // Emit the empty value back to parent
+      if (props.range || hasRelativeTimeframes.value) {
+        emit('changed', selectedRange.value)
+      } else {
+        emit('changed', '')
```

Comment:
I think this should emit the same object as all of the other emits; however, with empty values

Thoughts?

Reply:
It depends on whether the instance of this component is showing a single date / time, vs a range or relative time frames.  

If the former, we want to emit an empty string; 
else, it's an object with empty values for `start`, `end`, `timePeriodsKey`

Reply:
Pushed a commit.
How's this look for simplifying that check?

```
if (!props.range && showCalendar) {
  emit('changed', '')
} else {
  emit('changed', selectedRange.value)
}
```

The only time we need to emit an empty string (instead of a full Date string) is when
- the calendar is shown &&
- the calendar `range` attribute is set to `false`

Reply:
I think I get it, but regardless, can you document it? You're missing some sections in the markdown file for things like:

- **Slots** (if any)
- **Events** (see `/components/dropdown-menu.html#events` in the docs for an example).. since these events emit more complex data, you'll definitely want to show an example

Reply:
Done:
<img width="725" alt="Screen Shot 2022-09-08 at 12 45 23 PM" src="https://user-images.githubusercontent.com/103061463/189212283-e970ebf2-591f-4abb-94af-48a2993e9439.png">